src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingAssessmentResource.java

```diff
@@ -197,6 +204,17 @@ public ResponseEntity<Result> saveProgrammingAssessment(@PathVariable Long parti
                 || newManualResult.getParticipation().getExercise().getAssessmentDueDate().isBefore(ZonedDateTime.now()))) {
             messagingService.broadcastNewResult(newManualResult.getParticipation(), newManualResult);
         }
+
+        var isManualFeedbackRequest = programmingExercise.getAllowManualFeedbackRequests() && Objects.nonNull(participation.getIndividualDueDate())
+                && participation.getIndividualDueDate().isBefore(ZonedDateTime.now());
+        var isBeforeDueDate = Objects.nonNull(programmingExercise.getDueDate()) && programmingExercise.getDueDate().isBefore(ZonedDateTime.now());
+        if (isManualFeedbackRequest && isBeforeDueDate) {
+            participation.setIndividualDueDate(null);
+            studentParticipationRepository.save(participation);
+
+            programmingExerciseParticipationService.unlockStudentRepository(programmingExercise, (ProgrammingExerciseStudentParticipation) participation);
```

Comment:
**CR**: not covered by unit tests, but essential functionality.

Reply:
Shouldn't that test it? `ParticipationIntegrationTest.java:396-399`
```java
        assertThat(response.getIndividualDueDate()).isNotNull();
        assertThat(response.getIndividualDueDate()).isBefore(ZonedDateTime.now());

        verify(programmingExerciseParticipationService, times(1)).lockStudentRepository(programmingExercise, participation);
```